daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -1406,28 +1406,50 @@ private[lf] object DecodeV1 {
       BuiltinFunctionInfo(GREATER_DATE, BGreaterDate),
       BuiltinFunctionInfo(EQUAL, BEqual, minVersion = genMap),
       BuiltinFunctionInfo(EQUAL_LIST, BEqualList),
-      BuiltinFunctionInfo(EQUAL_CONTRACT_ID, BEqualContractId),
-      // FIXME https://github.com/digital-asset/daml/issues/3752
-      // Constrain max version of the following 'EQUAL_' builtin once
-      // generic equality is handled by the compiler
       BuiltinFunctionInfo(EQUAL_INT64, BEqual, implicitParameters = List(TInt64)),
       BuiltinFunctionInfo(
         EQUAL_DECIMAL,
         BEqualNumeric,
         maxVersion = Some(numeric),
         implicitParameters = List(TNat.Decimal)
       ),
-      BuiltinFunctionInfo(EQUAL_NUMERIC, BEqualNumeric, minVersion = numeric),
-      BuiltinFunctionInfo(EQUAL_TEXT, BEqual, implicitParameters = List(TText)),
-      BuiltinFunctionInfo(EQUAL_TIMESTAMP, BEqual, implicitParameters = List(TTimestamp)),
-      BuiltinFunctionInfo(EQUAL_DATE, BEqual, implicitParameters = List(TDate)),
-      BuiltinFunctionInfo(EQUAL_PARTY, BEqual, implicitParameters = List(TParty)),
-      BuiltinFunctionInfo(EQUAL_BOOL, BEqual, implicitParameters = List(TBool)),
+      BuiltinFunctionInfo(
+        EQUAL_NUMERIC,
```

Comment:
Why do we not map this to `BEqual` with an implicit parameter?

Reply:
Same as ``EQUAL_CONTRACT_ID``